### PR TITLE
[don't merge] Vector layers in CartoDB.js

### PR DESCRIPTION
**Pull request by fdansv**
Work in progress. Putting up this PR to track progressively.

@javisantana @rochoa @alonsogarciapablo

### CODE REVIEW

File: src/geo/leaflet/leaflet-vector-cartodb-group-layer-base.js [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+    // TODO: We can probably move this to an initialize method
+    this.model.bind('change:urls', function() {
+      self.__update(function() {
+        // if while the layer was processed in the server is removed
+        // it should not be added to the map
+        var id = L.stamp(self);
+        if (!map._layers[id]) {
+          return;
+        }
+        L.CartoDBd3Layer.prototype.onAdd.apply(self, [map]);
```
Comment (rochoa) [inline src/geo/leaflet/leaflet-vector-cartodb-group-layer-base.js]: This is a bit weird, isn't it?

File: src/geo/leaflet/leaflet-vector-cartodb-group-layer-base.js [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+        L.CartoDBd3Layer.prototype.setCartoCSS.apply(self, [self.model.collection.models[1].layers.models[0].get("cartocss")]);
```
Comment (rochoa) [inline src/geo/leaflet/leaflet-vector-cartodb-group-layer-base.js]: I'm totally lost here with `models[0]` and `models[1]` direct access. Can you elaborate a bit why it's like that :pray:

File: src/geo/leaflet/leaflet-vector-cartodb-group-layer-base.js [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+    } else {
+      self.error && self.error(err);
+      done && done();
+    }
+  },
+
+
+  _checkLayer: function() {
+    if (!this.options.added) {
+      throw new Error('the layer is not still added to the map');
```
Comment (rochoa) [inline src/geo/leaflet/leaflet-vector-cartodb-group-layer-base.js]: Wouldn't be better to return a Boolean? Not a fan of using exceptions for control flow.

File: src/geo/leaflet/leaflet-vector-cartodb-group-layer-base.js [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+      return this.options.featureOut && this.options.featureOut(o.e, o.layer);
+    }
+  },
+
+  /**
+   * Get the Leaflet Point of the event
+   * @params {Object} Map object
+   * @params {Object} Wax event object
+   */
+  _findPos: function (map, o) {
```
Comment (rochoa) [inline src/geo/leaflet/leaflet-vector-cartodb-group-layer-base.js]: Could we re-use this from `leaflet-cartodb-group-layer-base.js`?

File: src/geo/leaflet/vector.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+      cartocss: layerModel.get('cartocss') || layerModel.get('tile_style'),
+      named_map: layerModel.get('named_map'),
+      auth_token: layerModel.get('auth_token'),
+      no_cdn: layerModel.get('no_cdn'),
+      dynamic_cdn: layerModel.get('dynamic_cdn'),
+      loop: layerModel.get('loop') === false? false: true,
+      instanciateCallback: function() {
+        var cartocss = layerModel.get('cartocss') || layerModel.get('tile_style');
+
+        return '_cdbct_' + cdb.core.util.uniqueCallbackName(cartocss + query);
```
Comment (rochoa) [inline src/geo/leaflet/vector.js]: Use another name to not collide with torque's one (very unlikely due different cartocss most of the time).

File: src/geo/leaflet/vector.js [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    // Add CartoDB logo
+    if (this.options.cartodb_logo != false)
+      cdb.geo.common.CartoDBLogo.addWadus({ left:8, bottom:8 }, 0, map._container)
+  },
+
+  _getQuery: function(layerModel) {
+    var query = layerModel.get('query');
+    var qw = layerModel.get('query_wrapper');
+    if(qw) {
+      query = _.template(qw)({ sql: query || ('select * from ' + layerModel.get('table_name')) });
```
Comment (rochoa) [inline src/geo/leaflet/vector.js]: Do we want to keep support for `table_name`? Or should we make mandatory to have a SQL?

cc @javisantana

**Comment by fdansv** [comment]
@rochoa the majority of these comments apply to `leaflet-cartodb-group-layer-base.js`, as `leaflet-vector-cartodb-group-layer-base.js` is a modified copy of it.

File: src/geo/leaflet/leaflet-vector-cartodb-group-layer-base.js [hunk 7]
```diff
@@ -0,0 +1,306 @@
...
+    maxZoom: 30, // default leaflet zoom level for a layers is 18, raise it
+    extra_params:   {
+    },
+    cdn_url:        null,
+    subdomains:     null
+  },
+
+
+  initialize: function (options, layerModel, leafletMap) {
+    L.CartoDBd3Layer.prototype.initialize.apply(this, [options]);
```
Comment (alonsogarciapablo) [inline src/geo/leaflet/leaflet-vector-cartodb-group-layer-base.js]: Is this needed?

File: src/geo/leaflet/vector.js [hunk 8]
```diff
@@ -0,0 +1,95 @@
+var cdb = window.cdb;
```
Comment (alonsogarciapablo) [inline src/geo/leaflet/vector.js]: It seems like this file is not being used, right @fdansv?

Comment (fdansv) [inline src/geo/leaflet/vector.js]: That's right!

**Pull request closed without merging**